### PR TITLE
LVRUBY-10: Add 'Sign up' and 'Log in' functionality

### PR DESCRIPTION
**Pull request by olenaniemova**
## JIRA ticket

[LVRUBY-10](http://ssu-jira.softserveinc.com/secure/RapidBoard.jspa?rapidView=979&view=planning&selectedIssue=LVRUBY-10)

## Code reviewers

- [x] @PidtserkovnyiVitalii 
- [x] @vikabelous 

## Summary of issue

Add 'Sign up' and 'Log in' functionality. 

## Summary of change

Add new fields (fist_name, last_name, description) to User model. 
Add style and new fields to 'Sign up' and 'Log in' pages. 

## Review checklist

- [x] Rubocop doesn't raise any warnings and offenses
- [x] New features are covered with new specs
- [x] All specs pass

### CODE REVIEW

**Comment by vikabelous** [comment]
**Code Coverage:**
:green_heart: **100.0%** covered at **1.0** hits/line

File: app/views/devise/registrations/new.html.haml [hunk 1]
```diff
@@ -1,21 +1,53 @@
...
-    = f.label :password_confirmation
-    %br/
-    = f.password_field :password_confirmation, autocomplete: "new-password"
-  .actions
-    = f.submit "Sign up"
-= render "devise/shared/links"
+.container
+  .row.justify-content-center
+    .col-sm-5.text-center.pt-5
+      %p.h4.mb-2 Sign up
```
Comment (PidtserkovnyiVitalii) [inline app/views/devise/registrations/new.html.haml]: Place content on a new line after its parent element

**Comment by vikabelous** [comment]
**Code Coverage:**
:green_heart: **100.0%** covered at **1.0** hits/line

File: app/views/layouts/application.html.haml [hunk 2]
```diff
@@ -8,8 +8,10 @@
     = stylesheet_link_tag    'application', media: 'all', 'data-turbolinks-track': 'reload'
     = javascript_include_tag 'application', 'data-turbolinks-track': 'reload'
   %body
-    %p.notice
-      = notice
-    %p.alert
-      = alert
+    - unless notice.nil?
```
Comment (okohuch) [inline app/views/layouts/application.html.haml]: prefer the use of `if` over `unless` and check whether the object is present

File: app/views/layouts/application.html.haml [hunk 3]
```diff
@@ -8,8 +8,10 @@
...
+      %div.notice.alert.alert-info
+        = notice
+    - unless alert.nil?
```
Comment (okohuch) [inline app/views/layouts/application.html.haml]: prefer the use of `if` over `unless` and check whether the object is present

File: spec/factories/users.rb [hunk 4]
```diff
@@ -2,5 +2,10 @@
 
 FactoryBot.define do
   factory :user do
+    first_name { 'first name' }
+    last_name { 'last name' }
+    email { 'email@gmail.com' }
```
Comment (okohuch) [inline spec/factories/users.rb]: use a sequence for the email.

File: spec/factories/users.rb [hunk 5]
```diff
@@ -2,5 +2,10 @@
...
+    password { '123456a' }
+    description { '' }
```
Comment (okohuch) [inline spec/factories/users.rb]: what is the reason to have an empty description here?

File: spec/models/user_spec.rb [hunk 6]
```diff
@@ -3,4 +3,26 @@
 require 'rails_helper'
 
 RSpec.describe User, type: :model do
+  context 'create' do
+    it 'is valid' do
+      expect(build(:user)).to be_valid
+    end
+    it 'is not valid' do
```
Comment (okohuch) [inline spec/models/user_spec.rb]: add an empty line between blocks

File: spec/models/user_spec.rb [hunk 7]
```diff
@@ -3,4 +3,26 @@
...
+      expect(build(:user, password: '123')).not_to be_valid
+    end
+  end
+
+  context 'first_name and last_name are blank' do
+    it 'User is not valid' do
+      expect(build(:user, first_name: '')).not_to be_valid
+      expect(build(:user, first_name: '', last_name: '')).not_to be_valid
```
Comment (okohuch) [inline spec/models/user_spec.rb]: avoid more than one `expect` in the test

Comment (okohuch) [reply spec/models/user_spec.rb]: http://www.betterspecs.org/

**Comment by vikabelous** [comment]
**Code Coverage:**
:green_heart: **100.0%** covered at **1.0** hits/line

File: db/migrate/20190413091050_add_details_to_users.rb [hunk 8]
```diff
@@ -0,0 +1,9 @@
+# frozen_string_literal: true
+
+class AddDetailsToUsers < ActiveRecord::Migration[5.2]
+  def change
+    add_column :users, :first_name, :string, null: false
```
Comment (vikabelous) [inline db/migrate/20190413091050_add_details_to_users.rb]: Better use `change_table`
https://edgeguides.rubyonrails.org/active_record_migrations.html#changing-tables

File: spec/models/user_spec.rb [hunk 9]
```diff
@@ -3,4 +3,30 @@
...
+    it 'is valid' do
+      expect(build(:user)).to be_valid
+    end
+
+    it 'is not valid' do
+      expect(build(:user, password: '123')).not_to be_valid
+    end
+  end
+
+  context 'first_name and last_name are blank' do
```
Comment (vikabelous) [inline spec/models/user_spec.rb]: context block should start with 'when' or 'with' word
You should be able to read the `context` message + following `it` as a sentence, e.g. 'when first_name and last_name are blank it is not valid'

please read http://www.betterspecs.org/

File: spec/spec_helper.rb [hunk 10]
```diff
@@ -1,5 +1,8 @@
 # frozen_string_literal: true
 
+require 'simplecov'
```
Comment (vikabelous) [inline spec/spec_helper.rb]: SimpleCov is added to `rails_helper.rb`, please remove this

**Comment by vikabelous** [comment]
**Code Coverage:**
:green_heart: **100.0%** covered at **1.0** hits/line

**Comment by vikabelous** [comment]
**Code Coverage:**
:green_heart: **100.0%** covered at **1.0** hits/line

**Comment by vikabelous** [comment]
**Code Coverage:**
:green_heart: **100.0%** covered at **1.0** hits/line

File: spec/models/user_spec.rb [hunk 11]
```diff
@@ -3,4 +3,36 @@
...
+  end
+
+  context 'when user\'s data are not correct' do
+    it 'is not valid' do
+      expect(build(:user, password: '123')).not_to be_valid
+    end
+  end
+end
+
+RSpec.describe 'validations', type: :model do
```
Comment (vikabelous) [inline spec/models/user_spec.rb]: You should have only one top level `RSpec.describe` statement
You can have more `describe` blocks inside that top level `describe` same way you have multiple `context` blocks.

File: spec/models/user_spec.rb [hunk 12]
```diff
@@ -3,4 +3,36 @@
 require 'rails_helper'
 
 RSpec.describe User, type: :model do
+  context 'when user\'s data are correct' do
```
Comment (vikabelous) [inline spec/models/user_spec.rb]: Use double quotes if you need a single quote inside your string

**Comment by vikabelous** [comment]
**Code Coverage:**
:green_heart: **100.0%** covered at **1.0** hits/line

**Comment by vikabelous** [comment]
**Code Coverage:**
:green_heart: **100.0%** covered at **1.0** hits/line

**Comment by vikabelous** [comment]
**Code Coverage:**
:green_heart: **100.0%** covered at **1.0** hits/line

**Pull request merged**